codex/sales/reservations.nim

```diff
@@ -95,12 +99,26 @@ const
   SalesKey = (CodexMetaKey / "sales").tryGet # TODO: move to sales module
   ReservationsKey = (SalesKey / "reservations").tryGet
 
+proc hash*(x: AvailabilityId): Hash {.borrow.}
 proc all*(self: Reservations, T: type SomeStorableObject): Future[?!seq[T]] {.async.}
 
+template withLock(lock, body) =
+  try:
+    trace "Acquiring lock"
+    await lock.acquire()
+    trace "Lock acquired"
+    body
+  finally:
+    if lock.locked:
+      trace "Releasing lock"
+      lock.release()
+      trace "Lock released"
```

Comment:
Hmm, I think and I hope we don't really need these 4 trace statements. Since I'm frequently running lots of cluster tests and looking at GBs of log files already, I propose to remove them until I'm proven wrong.

Reply:
I see your point. Would scoping those lock logs to topic `locks` be enough for you to filter them out?

Reply:
That could work, but mostly when I'm using the logs I don't really need detail down the individual locks and operations. It's much more useful to see what is being done instead of the details of how.

Reply:
I will remove it for now. These types of locks are usually very handy when you are trying to detect dead-locks though...

Also, I completely understand the pain of the logs, but IMHO this should be handled with proper log levels and separation into topics, that can be filtered out based on the need. We might not be using these appropriately that much. For example for "It's much more useful to see what is being done instead of the details of how." IMHO the `info` level should be used, while `trace` should be more low-level implementation notification (like doing locks).